### DISCUSSION TITLE
Annotation Query with Template Variable?

### DISCUSSION THREAD

**Discussion by bangau1** [grafana/grafana#43354] [Dashboards, Panels & Visualizations]
Hi,

I am wondering whether there is already template variable support within the annotation query? I tried to incorporate template variable in my annotation query, but it doesn't work.

For example:
- I define 1 template variable called `app`. 
- Then I only want to show annotation from loki data source with query LogQL something like this: `{k8s_cluster="cluster", namespace="prod", app="$app"} |= "[ERROR]"`, but somehow this doesn't show any annotation at all. But when I hardcoded the $app value directly in the query, it works.

Any advice? Or perhaps this should be categorized as feature request?

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
